engine/order_manager.go

```diff
@@ -388,6 +388,34 @@ func (m *OrderManager) GetOrdersSnapshot(s order.Status) ([]order.Detail, time.T
 	return os, latestUpdate
 }
 
+// GetOrdersFiltered returns a snapshot of all orders in the order store.
+// Filtering is applied based on the order.Filter unless entries are empty
+func (m *OrderManager) GetOrdersFiltered(f *order.Filter) ([]order.Detail, time.Time, error) {
+	if m == nil {
+		return nil, time.Time{}, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+
+	if m == nil || atomic.LoadInt32(&m.started) == 0 {
+		return nil, time.Time{}, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+	var os []order.Detail
+	var latestUpdate time.Time
+	for _, v := range m.orderStore.Orders {
```

Comment:
out of scope: might be better fixing up (make a new map, don't ref any orders to the underlying map) and using m.orderStore.get() method. Let me know what you think.

Reply:
I moved the filtering to the order store instead.

Reply:
Changes look awesome! Will mark this for review.